client/src/components/getCompareListsData.js

```diff
@@ -0,0 +1,50 @@
+import PropTypes from "prop-types";
+
+function getCompareListsData({
```

Comment:
This function is very complex and you access React states functions inside.
I suggest reading "One function – one action" in https://javascript.info/function-basics#function-naming.

This function doesn't get anything, it sets something. 
Instead, try to calculate the data and return it. In your React component, you can read the result and update your states.

In addition, this is not a component. Thus, I wouldn't place it next to the components and move it to a different folder. 
The prop-types shouldn't be required for the same reason.

Reply:
Hey, many thanks for your input. With my best intend I tried fixing it and renamed and re-structured my functions. Would you mind having a look again?